sdk/lib/crt/stdio/stat64.c

```diff
@@ -68,10 +67,13 @@ int CDECL _tstat64(const _TCHAR *path, struct __stat64 *buf)
   {
     mode |= _S_IFREG;
     /* executable? */
-    if (plen > 6 && path[plen-4] == '.')  /* shortest exe: "\x.exe" */
+    if (plen > 6 && path[plen-4] == __T('.'))  /* shortest exe: "\x.exe" */
     {
-      unsigned int ext = tolower(path[plen-1]) | (tolower(path[plen-2]) << 8) |
-                                 (tolower(path[plen-3]) << 16);
+
+      TCHAR4 ext = (TCHAR4)_totlower(path[plen-1])
+                   | ((TCHAR4)_totlower(path[plen-2]) << TCSIZE_BITS)
+                   | ((TCHAR4)_totlower(path[plen-3]) << 2*TCSIZE_BITS);
+
       if (ext == EXE || ext == BAT || ext == CMD || ext == COM)
```

Comment:
Using a _tcsnicmp call with the strings L".exe", etc... L".com" would possibly be more readable.

Reply:
This is based on Wine code, which does exactly this shift stuff

Reply:
Is there a reason why it was done this way?

Reply:
Well, this code appears to be at least 16 years old. So even if there was a reason, it's probably not valid anymore.
That said, it's usually best for us to keep the code in sync with Wine unless we have a specific reason not to. That means less maintenance effort for us.